datasources-agroal/src/main/java/org/wildfly/extension/datasources/agroal/DataSourceService.java

```diff
@@ -115,7 +126,7 @@ public void start(StartContext context) throws StartException {
 
         if (jta || xa) {
             TransactionManager transactionManager = ContextTransactionManager.getInstance();
-            TransactionSynchronizationRegistry transactionSynchronizationRegistry = transactionSynchronizationRegistryInjector.getValue();
+            TransactionSynchronizationRegistry transactionSynchronizationRegistry = transactionSynchronizationRegistrySupplier.get();
```

Comment:
I think transactionSynchronizationRegistrySupplier can be null if jta is false.

Reply:
Correct - thanks for spotting this out ! It is fixed now.